<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>816</width>
    <height>929</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Длина резца (мм):</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cutter_len">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>10</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>351</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Длина материала (мм) и кол-во:</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>351</width>
      <height>285</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLineEdit" name="line1"/>
     </item>
     <item row="2" column="1">
      <widget class="QSpinBox" name="n3"/>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="n1"/>
     </item>
     <item row="5" column="0">
      <widget class="QLineEdit" name="line6"/>
     </item>
     <item row="4" column="1">
      <widget class="QSpinBox" name="n5"/>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="n4"/>
     </item>
     <item row="4" column="0">
      <widget class="QLineEdit" name="line5"/>
     </item>
     <item row="5" column="1">
      <widget class="QSpinBox" name="n6"/>
     </item>
     <item row="7" column="0">
      <widget class="QLineEdit" name="line8"/>
     </item>
     <item row="6" column="0">
      <widget class="QLineEdit" name="line7"/>
     </item>
     <item row="8" column="0">
      <widget class="QLineEdit" name="line9"/>
     </item>
     <item row="6" column="1">
      <widget class="QSpinBox" name="n7"/>
     </item>
     <item row="1" column="0">
      <widget class="QLineEdit" name="line2"/>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="n2"/>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="line3"/>
     </item>
     <item row="3" column="0">
      <widget class="QLineEdit" name="line4"/>
     </item>
     <item row="9" column="0">
      <widget class="QLineEdit" name="line10"/>
     </item>
     <item row="7" column="1">
      <widget class="QSpinBox" name="n8"/>
     </item>
     <item row="8" column="1">
      <widget class="QSpinBox" name="n9"/>
     </item>
     <item row="9" column="1">
      <widget class="QSpinBox" name="n10"/>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>90</y>
      <width>351</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Длины заготовок (мм) и кол-во:</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>120</y>
      <width>351</width>
      <height>285</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="9" column="0">
      <widget class="QLineEdit" name="line20"/>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="line13"/>
     </item>
     <item row="3" column="0">
      <widget class="QLineEdit" name="line14"/>
     </item>
     <item row="7" column="1">
      <widget class="QSpinBox" name="n18"/>
     </item>
     <item row="8" column="1">
      <widget class="QSpinBox" name="n19"/>
     </item>
     <item row="0" column="0">
      <widget class="QLineEdit" name="line11"/>
     </item>
     <item row="2" column="1">
      <widget class="QSpinBox" name="n13"/>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="n11"/>
     </item>
     <item row="4" column="1">
      <widget class="QSpinBox" name="n15"/>
     </item>
     <item row="6" column="0">
      <widget class="QLineEdit" name="line17"/>
     </item>
     <item row="4" column="0">
      <widget class="QLineEdit" name="line15"/>
     </item>
     <item row="5" column="1">
      <widget class="QSpinBox" name="n16"/>
     </item>
     <item row="7" column="0">
      <widget class="QLineEdit" name="line18"/>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="n14"/>
     </item>
     <item row="6" column="1">
      <widget class="QSpinBox" name="n17"/>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="n12"/>
     </item>
     <item row="8" column="0">
      <widget class="QLineEdit" name="line19"/>
     </item>
     <item row="1" column="0">
      <widget class="QLineEdit" name="line12"/>
     </item>
     <item row="9" column="1">
      <widget class="QSpinBox" name="n20"/>
     </item>
     <item row="5" column="0">
      <widget class="QLineEdit" name="line16"/>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="button">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>180</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Calculate</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>0</y>
      <width>221</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>7</pointsize>
     </font>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600; font-style:italic; text-decoration: underline;&quot;&gt;Created by Ru&amp;amp;Ko&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600; font-style:italic; text-decoration: underline;&quot;&gt;All rights reserved&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600; font-style:italic; text-decoration: underline;&quot;&gt;Version: 1.0.0 2022&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="output">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>800</width>
      <height>512</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="toPDF">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>250</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>to PDF</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clear">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>310</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Имя файла для сохранения в PDF:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="file_name">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>50</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>result.pdf</string>
    </property>
   </widget>
   <widget class="QPushButton" name="help">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>370</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Help</string>
    </property>
   </widget>
   <widget class="QPushButton" name="txt">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>120</y>
      <width>101</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Load from TXT</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="../../Downloads/knife-transparent-anime-8.qrc"/>
 </resources>
 <connections/>
</ui>
